tick/inference/tests/robust_linear_regression_test.py

```diff
@@ -60,53 +57,48 @@ def test_RobustLinearRegression_fit(self):
         """...Test RobustLinearRegression fit with different solvers and penalties
         """
         X, y, weights0, interc0, sample_intercepts0 = self.get_train_data(True)
-        for i, (solver, penalty) \
-                in enumerate(product(RobustLinearRegression._solvers.keys(),
-                                     RobustLinearRegression._penalties.keys())):
+
+        solvers = RobustLinearRegression._solvers.keys()
+        for i, solver in enumerate(solvers):
             learner_keywords = {
                 'C_sample_intercepts': Test.n_samples / Test.noise_level,
                 'fit_intercept': True, 'fdr': Test.target_fdr,
-                'max_iter': 100, 'tol': 1e-10, 'solver': solver,
-                'penalty': penalty, 'verbose': False
+                'max_iter': 3000, 'tol': 1e-7, 'solver': solver,
+                'penalty': 'none', 'verbose': False
             }
-            if penalty != 'none':
-                learner_keywords['C'] = 1e8
+
             learner = RobustLinearRegression(**learner_keywords)
             learner.fit(X, y)
 
-            weights = [1.70872603, 1.68487097, 1.92782337, 2.55855144,
-                       3.5299368]
-            interc = -2.77864308544
-            fdp_ = 0.206349206349
+            weights = [1.82145051, 2.32011366, 2.6886905, 2.53289584,
+                       2.86991904]
+            interc = -2.9877245464563931
+            fdp_ = 0.23076923076923078
             power = 1.0
 
             np.testing.assert_array_almost_equal(weights, learner.weights, 2)
-            self.assertAlmostEqual(interc, learner.intercept, 3)
+            self.assertAlmostEqual(interc, learner.intercept, 2)
             self.assertAlmostEqual(fdp_, support_fdp(sample_intercepts0,
-                                                     learner.sample_intercepts), 4)
+                                                     learner.sample_intercepts),
+                                   4)
             self.assertAlmostEqual(power,
                                    support_recall(sample_intercepts0,
                                                   learner.sample_intercepts), 4)
 
         X, y, weights0, interc0, sample_intercepts0 = self.get_train_data(False)
-        for i, (solver, penalty) \
-                in enumerate(product(RobustLinearRegression._solvers.keys(),
-                                     RobustLinearRegression._penalties.keys())):
+        for i, solver in enumerate(solvers):
             learner_keywords = {
                 'C_sample_intercepts': Test.n_samples / Test.noise_level,
                 'fit_intercept': False, 'fdr': Test.target_fdr,
-                'max_iter': 100, 'tol': 1e-10, 'solver': solver,
-                'penalty': penalty, 'verbose': False
+                'max_iter': 3000, 'tol': 1e-7, 'solver': solver,
```

Comment:
How long takes this test to run? I think test should be < 300 ms if possible and 2sec in the worst case.